### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by beniaminopozzan**
The README.md was quite outdated and was still mentioning `px4_ros_com`.
Now the files is updated with the current PX4 message structure and advises the users about the different branches and their pairing with PX4 and ROS versions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,25 +2,39 @@
 
 [![GitHub license](https://img.shields.io/github/license/PX4/px4_msgs.svg)](https://github.com/PX4/px4_msg/blob/master/LICENSE) [![Build package](https://github.com/PX4/px4_msgs/workflows/Build%20package/badge.svg)](https://github.com/PX4/px4_msgs/actions)
 
-This package contains the ROS2 message definitions of the [PX4 Pro ecosystem](https://px4.io/). Building this package generates all the required interfaces to interface ROS2 nodes with the PX4 autopilot internals, which use the [uORB messaging API](https://dev.px4.io/en/middleware/uorb.html). Currently the messages of this package represent a dependency to [`px4_ros_com` package](https://github.com/PX4/px4_ros_com).
+This interface package contains the ROS 2 message definitions of the [PX4 Pro ecosystem](https://px4.io/). Building this package generates all the required interfaces to interface ROS 2 nodes with the PX4 autopilot internals.
```
Comment (mrpollo) [inline README.md:5]: We are no longer using "PX4 Pro" 

```suggestion
ROS 2 message definitions for the [PX4 Autopilot](https://px4.io/) project.

Building this package generates all the required interfaces to interface ROS 2 nodes with the PX4 internals.
```

File: README.md [hunk 2]
```diff
@@ -2,25 +2,39 @@
...
 
-## uORB message definitions
+## Supported versions and compatibility
 
-The uORB message definitions, which represent the counter-part of the ROS2 messages found in this package, can be found on the [PX4 Firmware repository](https://github.com/PX4/Firmware).
+Depending on the PX4 and ROS versions you want to use, you need to checkout the appropriate branch of this package:
 
-## How are these messsage definitions generated?
+| PX4 version   | ROS 2 versions          | branch                                                            |
```
Comment (mrpollo) [inline README.md:11]: I think we should also add the Ubuntu version we support for each @beniaminopozzan. Would that be possible?

File: README.md [hunk 3]
```diff
@@ -2,25 +2,39 @@
...
+| `main` branch | Foxy - Humble - Rolling | [main](https://github.com/PX4/px4_msgs)                           |
 
-When uORB message definitions in the [PX4 firmware](https://github.com/PX4/Firmware) repository change, a CI/CD pipeline automatically generates and pushes updated ROS message definitions to this repository. The ROS message definitions are generated by the [`uorb_to_ros_msgs.py`](https://github.com/PX4/Firmware/blob/master/msg/tools/uorb_to_ros_msgs.py) script. One can also use this script to generate its own ROS message definitions for new or modified uORB messages.
+### Manual messages synchronization
+
+When PX4 message definitions in the `main` branch of [PX4 Autopilot](https://github.com/PX4/Autopilot) change, a CI/CD pipeline automatically copies and pushes updated ROS message definitions to this repository. This ensures that this repository `main` branch and the PX4-Autopilot `main` branch are always up to date.
+However, if you are using a custom PX4 version and you modified existing messages or created new one, then you have to manually synchronize them in this repository:
+
```
Comment (mrpollo) [inline README.md:21]:
```suggestion
### Manual Message Sync

```

File: README.md [hunk 4]
```diff
@@ -2,25 +2,39 @@
...
 
-## How are these messsage definitions generated?
+| PX4 version   | ROS 2 versions          | branch                                                            |
+|---------------|-------------------------|-------------------------------------------------------------------|
+| 1.13          | Foxy                    | [release/1.13](https://github.com/PX4/px4_msgs/tree/release/1.13) |
+| 1.14          | Foxy - Humble - Rolling | [release/1.14](https://github.com/PX4/px4_msgs/tree/release/1.14) |
+| `main` branch | Foxy - Humble - Rolling | [main](https://github.com/PX4/px4_msgs)                           |
 
-When uORB message definitions in the [PX4 firmware](https://github.com/PX4/Firmware) repository change, a CI/CD pipeline automatically generates and pushes updated ROS message definitions to this repository. The ROS message definitions are generated by the [`uorb_to_ros_msgs.py`](https://github.com/PX4/Firmware/blob/master/msg/tools/uorb_to_ros_msgs.py) script. One can also use this script to generate its own ROS message definitions for new or modified uORB messages.
+### Manual messages synchronization
```
Comment (mrpollo) [inline README.md:17]:
```suggestion
### Messages Sync from PX4
```

File: README.md [hunk 5]
```diff
@@ -2,25 +2,39 @@
...
+
+When PX4 message definitions in the `main` branch of [PX4 Autopilot](https://github.com/PX4/Autopilot) change, a CI/CD pipeline automatically copies and pushes updated ROS message definitions to this repository. This ensures that this repository `main` branch and the PX4-Autopilot `main` branch are always up to date.
```
Comment (mrpollo) [inline README.md:19]: Consider adding a link to the CI workflow and actions.

* [Workflow](https://github.com/PX4/PX4-Autopilot/blob/main/.github/workflows/metadata.yml#L119) 
* [Actions](https://github.com/PX4/PX4-Autopilot/actions/workflows/metadata.yml)

File: README.md [hunk 6]
```diff
@@ -2,25 +2,39 @@
...
 
 ## Bug tracking and feature requests
 
 Use the [Issues](https://github.com/PX4/px4_msgs/issues) section to create a new issue. Report your issue or feature request [here](https://github.com/PX4/px4_msgs/issues/new).
 
 ## Questions and troubleshooting
 
-Reach the PX4 development team on the `#messaging` PX4 Slack channel:
-[![Slack](https://px4-slack.herokuapp.com/badge.svg)](http://slack.px4.io)
+Reach the PX4 development team on the PX4 Discord Server:
```
Comment (mrpollo) [inline README.md:39]: Can you please move the badge to the top of the readme? I think its best if it matches what we have on PX4 Autopilot repo.

**Review by mrpollo — CHANGES_REQUESTED** [review]
Hey @beniaminopozzan thanks for taking the time, I really appreciate the effort. I hope you will consider the proposals I sent over. I look forward to reading your comments.

File: README.md (hunk 1 above)
Comment (beniaminopozzan) [reply README.md:5]: I see, thanks!

File: README.md (hunk 2 above)
Comment (beniaminopozzan) [reply README.md:15]: Yes, done!

File: README.md (hunk 5 above)
Comment (beniaminopozzan) [reply README.md]: Yes! I added the Workflow link.

File: README.md (hunk 6 above)
Comment (beniaminopozzan) [reply README.md]: Done!

File: README.md (hunk 2 above)
Comment (mrpollo) [reply README.md:15]: Would it be possible to use this format?

```
| ROS     | PX4    | Ubuntu       |
| ------- | ------ | ------------ |
| Foxy    | [v1.13](https://github.com/PX4/px4_msgs/tree/release/1.13)  | Ubuntu 20.04 |
| Humble  | ≥ [v1.14](https://github.com/PX4/px4_msgs/tree/release/1.14) | Ubuntu 22.04 |
| Rolling | ≥ [v1.14](https://github.com/PX4/px4_msgs/tree/release/1.14) | Ubuntu 22.04 |
```

Comment (beniaminopozzan) [reply README.md:15]: This exclude the combination Foxy + v1.14, I can expand on this though:


| PX4            | ROS 2   | Ubuntu       | branch                                                            |
|----------------|---------|--------------|-------------------------------------------------------------------|
| [v1.13](https://github.com/PX4/px4_msgs/tree/release/1.13)           | Foxy    | Ubuntu 20.04 | [release/1.13](https://github.com/PX4/px4_msgs/tree/release/1.13) |
| [v1.14](https://github.com/PX4/px4_msgs/tree/release/1.14)           | Foxy    | Ubuntu 20.04 | [release/1.14](https://github.com/PX4/px4_msgs/tree/release/1.14) |
| [v1.14](https://github.com/PX4/px4_msgs/tree/release/1.14)           | Humble  | Ubuntu 22.04 | [release/1.14](https://github.com/PX4/px4_msgs/tree/release/1.14) |
| [v1.14](https://github.com/PX4/px4_msgs/tree/release/1.14)           | Rolling | Ubuntu 22.04 | [release/1.14](https://github.com/PX4/px4_msgs/tree/release/1.14) |
| [main](https://github.com/PX4/px4_msgs/tree/main)  | Foxy    | Ubuntu 22.04 | [main](https://github.com/PX4/px4_msgs)                           |
|[main](https://github.com/PX4/px4_msgs/tree/main)  | Humble  | Ubuntu 22.04 | [main](https://github.com/PX4/px4_msgs)                           |
| [main](https://github.com/PX4/px4_msgs/tree/main)   | Rolling | Ubuntu 22.04 | [main](https://github.com/PX4/px4_msgs)                           |

What do you think?
I cannot put "≥ v1.14" as v1.14 (release/1.14) and `main` have already diverged in term off messages definitions.

**Pull request merged**